docs/developer/automerge.md

```diff
@@ -0,0 +1,25 @@
+# Automatic Merging of Pull Requests in OSM
+This document aims to explain how OSM utilizes the [automerge-action bot](https://github.com/pascalgn/automerge-action) to automatically merge pull requests and how it was set up.
+
+## Workaround to triggering workflows
+By default the automerge-action bot uses a `GITHUB_TOKEN` to perform automerging; however, events triggered via `GITHUB_TOKEN` will not further trigger additional workflow runs. To work around this, we created a GitHub App (Generate_Token) to generate a token that is then used to create a JSON Web Token (JWT) that authenticates and retrieves the Installation Access Token (IAT) from the GitHub API. Passing in the IAT as the `GITHUB_TOKEN`, will allow both `on:push` and `on:pull_request` workflows to trigger.
+
+## How it works
+- Automerge only occurs when the `automerge` label is assigned. To prevent automerging, you can either not assign the `automerge` label or include one of the following labels: `wip` or `do-not-merge/hold`.
+- Once labelled with `automerge` and neither of the blocking labels, a merge attempt will trigger only if all required status checks have passed (status checks are designated as required in the branch protection rules) and the required number of review approvals have been given.
+- If the pull request is not up to date, the automerge bot will automatically attempt to rebase. If it fails to rebase it will exit with error code 1 and the user must resolve the conflicts. If the rebase is successful, then the status checks will be re-triggered and once the required checks have passed the pull request will be automatically merged.
```

Comment:
is it possible to keep only rebase/merge with main functionality but disable automerge to start with?

Reply:
I think changing the `UPDATE_LABELS` to be like `rebase` would do it. Then PRs labelled with `rebase` should only rebase and   if the `automerge` label is added then it'll automerge, but let me test it out and I'll let you know!

Reply:
@shalier @sozercan why is there a need for additional labels like rebase?

Reply:
Confirming that changing the `UPDATE_LABELS` to `rebase` will let the PR rebase without automerging

Reply:
@shashankram the new `rebase` label would only be added if we wanted to only rebase PRs but not automerge them.

Reply:
Got it, that's a good feature to have!

Reply:
if what's needed is a bot that constantly rebases instead of automerges (which also eliminates optional checks), I think that's a better approach to start with.

Reply:
@sozercan, @shashankram It turns out that setting the `UPDATE_LABELS` to `rebase` would mean that a PR will only rebase with the `rebase` label and that the `automerge` label would not rebase. And adding the `automerge` label in addition to the `rebase` label to `UPDATE_LABELS` wouldn't work as it then means in order to rebase you'd need both the `automerge` and `rebase` labels. For now, I've made it so that the `automerge` label will do both rebasing and automerging.